### PR TITLE
fix pthread_cond_timedwait

### PR DESCRIPTION
**Pull request by sgaragagghu**

### CODE REVIEW

File: client/transport.c [hunk 1]
```diff
@@ -353,12 +353,15 @@ static int do_request(int type, xtime_t start_time, xtime_t end_time, void *data
...
 	while(!reply_rcvd){
-		if(pthread_cond_timedwait(&g_process_cond, &g_process_mutex, &expire_time)) {
+		errno = 0;
+		if((ret = pthread_cond_timedwait(&g_process_cond, &g_process_mutex, &expire_time))) {
 			last_reply_id = reply_id_private;
 			reply_data = NULL;
 			reply_callback = NULL;
-			if(errno != EAGAIN || errno != ETIMEDOUT) {
-				xdag_err("pthread_cond_timedwait failed [function: do_request]");
+			if(errno && (errno != EAGAIN || errno != ETIMEDOUT)) {
```
Comment (jonano614) [inline client/transport.c]: Do need || here? (errno != EAGAIN || errno != ETIMEDOUT) is always true.

Comment (sgaragagghu) [reply client/transport.c]: how can be always true? errno may be EAGAIN for example, or EINVAL
on some system errno is set and in others err code is returned from the function so i check both

Comment (sgaragagghu) [reply client/transport.c]: I removed it anyway since by standard error number whould be returned from the function

File: client/transport.c [hunk 2]
```diff
@@ -353,12 +353,12 @@ static int do_request(int type, xtime_t start_time, xtime_t end_time, void *data
...
 
 	while(!reply_rcvd){
-		if(pthread_cond_timedwait(&g_process_cond, &g_process_mutex, &expire_time)) {
+		if((ret = pthread_cond_timedwait(&g_process_cond, &g_process_mutex, &expire_time))) {
 			last_reply_id = reply_id_private;
 			reply_data = NULL;
 			reply_callback = NULL;
-			if(errno != EAGAIN || errno != ETIMEDOUT) {
-				xdag_err("pthread_cond_timedwait failed [function: do_request]");
+			if(ret != EAGAIN || ret != ETIMEDOUT) {
```
Comment (jonano614) [inline client/transport.c]: `ret != EAGAIN || ret != ETIMEDOUT` is always true
You should use `ret != EAGAIN && ret != ETIMEDOUT`

Comment (sgaragagghu) [reply client/transport.c]: wow, yes fixed!

**Pull request merged**